src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -271,9 +271,6 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
             return errorResponse;
         }
 
-        // Only save after checking for errors
```

Comment:
This seems like a bug to me - we use the tenplate/solution participations from the gui to update the participations in the database.
The problem is that exercises and participations have a circular structure:
An exercise has a templateParticipation, the templateParticipation the exercise attached to it. This is why the templateParticipation doesn't include it's exercise, setting the entry's exercise id in the participation table to null.

I don't see any reason, why we should save the participation data when updating a programming-exercise.